concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/TreeStateTest.hs

```diff
@@ -0,0 +1,908 @@
+          ruiTimeout = 0,
+          ruiPayload = MicroGTUPerEuroUpdatePayload 2
+        }
+
+dummyUpdateInstruction :: UpdateSequenceNumber -> UpdateInstruction
+dummyUpdateInstruction usn =
+    makeUpdateInstruction
+        (dummyRawUpdateInstruction usn)
+        (Map.fromList [(0, dummySigSchemeKeys)])
+
+dummyUpdateInstructionWM :: UpdateSequenceNumber -> WithMetadata UpdateInstruction
+dummyUpdateInstructionWM usn =
+    addMetadata ChainUpdate 0 $ dummyUpdateInstruction usn
+
+dummyChainUpdate :: UpdateSequenceNumber -> BlockItem
+dummyChainUpdate usn = chainUpdate $ dummyUpdateInstructionWM usn
+
+-- |A valid 'AccountCreation' with expiry 1596409020
+dummyAccountCreation :: AccountCreation
+dummyAccountCreation = readAccountCreation . BSL.fromStrict $ $(makeRelativeToProject "testdata/transactionverification/verifiable-credential.json" >>= embedFile)
+
+credentialDeploymentWM :: WithMetadata AccountCreation
+credentialDeploymentWM = addMetadata CredentialDeployment 0 dummyAccountCreation
+
+dummyCredentialDeployment :: BlockItem
+dummyCredentialDeployment = credentialDeployment credentialDeploymentWM
+
+testLookupLiveTransaction :: Spec
+testLookupLiveTransaction = describe "lookupLiveTransaction" $ do
+    it "present" $ do
+        assertEqual
+            "status transaction 1"
+            (Just $ Received 0 (dummySuccessTransactionResult 1))
+            $ lookupLiveTransaction (th 1) sd
+        assertEqual
+            "status transaction 2"
+            (Just $ Received 0 (dummySuccessTransactionResult 2))
+            $ lookupLiveTransaction (th 2) sd
+        assertEqual
+            "status transaction 3"
+            (Just $ Received 0 (dummySuccessTransactionResult 3))
+            $ lookupLiveTransaction (th 3) sd
+    it "absent"
+        $ assertEqual
+            "status transaction 4"
+            Nothing
+        $ lookupLiveTransaction (th 4) sd
+  where
+    th n = getHash (dummyTransactionBI n)
+    addTrans n = snd . addTransaction (dummyTransactionBI n) 0 (dummySuccessTransactionResult n)
+    sd =
+        dummyInitialSkovData
+            & transactionTable
+                %~ addTrans 1
+                . addTrans 2
+                . addTrans 3
+
+testLookupTransaction :: Spec
+testLookupTransaction = describe "lookupTransaction" $ do
+    it "finalized" $ lu (th 1) (Just $ Finalized $ FinalizedTransactionStatus 1 0)
+    it "live" $ lu (th 2) (Just $ Live $ Received 0 $ dummySuccessTransactionResult 2)
+    it "absent" $ lu (th 5) Nothing
+  where
+    th n = getHash (dummyTransactionBI n)
+    addTrans n = snd . addTransaction (dummyTransactionBI n) 0 (dummySuccessTransactionResult n)
+    sd =
+        dummyInitialSkovData
+            & transactionTable
+                %~ addTrans 2
+                . addTrans 3
+    db =
+        (lldbWithGenesis @'P6)
+            { lldbTransactions = HM.fromList [(th 1, FinalizedTransactionStatus 1 0)]
+            }
+    lu hsh expect = do
```

Comment:
Please avoid abbreviations

Reply:
lookup, hash and expectation. I think it's ok and clear from the context? Not sure a longer variable name would make it easier to judge what it does. It's a local bind within a local context where things are pretty clear I think.

Reply:
`lu` is not clear to be "lookup".  It would really not cost you anything to call this `lookupAndCheck` or some such.

I don't get this aversion.